routing.asciidoc

```diff
@@ -82,304 +99,201 @@ As Wei gets message from the escrow that Bob has deposited the 10 golden coins W
+
+The payment should _only_ be able to be redeemed if a valid solution is revealed. Critical, all conditions need to be timed in order to allow the construct to return the funds back to the sender if a solution to this condition isn't revealed.
+The combination of the condition, and a timeout on the condition gives the payment a trait we commonly refer to as atomicity: either the payment happens, or the receiver if refunded the funds.
+
+=== Conditional Chained Payment ===
+
+Building upon our conditional payment, it may be possible to *chain* this payment, allowing it to involve the payer, the payee, and possibly several intermediaries. 
+Each intermediary, is able to present a _slightly_ modified version of the condition (without invalidating it all together), and so so in an iterated manner until the conditional payment reaches the payee.
+Once it reaches the payee, then the payment should be able to be _iteratively_ resolved, starting at the payee all the way back to the payer. 
+
+Each chaining creates an "incoming" and "outgoing" conditional payment.
+A node receives a conditional payment from a party (incoming condition), and then extends the conditional payment to the next party in the chain (outgoing condition).
+The payment is extended in from payer to payee, but settled from payee to payer, as each of the intermediaries gain the solution to the outgoing condition, and use that (possibly augmenting it) to satisfy the incoming solution.
+
+Typically the payer rewards the intermediaries by sending slightly more than the payment amount, in order to allow the intermediaries to send out less with their outgoing payment than what they received from the incoming payment.
+The difference between these two payment values makes up the "forwarding fee" collected by the intermediary.
+
+=== Conditional Chained End to End Secure Payment === 
+
+With our final addition, we'll achieve "end to end security".
+By this we mean that: no intermediaries are able to "claim" the payment without first obtaining the solution from someone further down from them in the chain.
+Additionally, we also require that the amount the payer intended to send is fully received by the payee.
+Finally, we require that non of the intermediaries are able to "contaminate" the payment beyond giving incorrect directions to the party that directly follows them.
+In other words, the intermediary shouldn't be able to materially affect the propagation of the payment several hops away from it.
+
+== Hash Time Locked Contracts ==
+
+In this section, we'll construct a conditional chained end to end payment known as the HTLC. 
+At each step we'll add a new component, then examine it in light of our original definition to ensure all requirements and security properties are reached.
+
+First, the "condition". For an HTLC, the condition is typically the reveal of a hash pre-image that matches a particular hash.
```

Comment:
the following could be an ordered list. they are usually good for learners as it is like "5 steps to xyz"

Reply:
The rest of the paragraph isn't a series of items though, unless you're referring to this section as a whole?